src/storage/txn/commands/mod.rs

```diff
@@ -451,3 +585,25 @@ impl Debug for Command {
         self.command_ext().fmt(f)
     }
 }
+
+pub trait ReadCommand<S: Snapshot>: CommandExt {
+    fn process_read(self, snapshot: S, statistics: &mut Statistics) -> Result<ProcessResult>;
+}
+
+pub struct WriteContext<'a> {
```

Comment:
I'm thinking about whether it's a good idea to put other dependencies (LockMgr and PdClient) of write commands to WriteContext. Then, when we are adding or removing these dependencies, we'll need to only modify one place instead of changing all the command files.
What do you think?

Reply:
> I'm thinking about whether it's a good idea to put other dependencies (LockMgr and PdClient) of write commands to WriteContext. Then, when we are adding or removing these dependencies, we'll need to only modify one place instead of changing all the command files.
> What do you think?

@sticnarf PTAL.
I tried it and it seems good.